### DISCUSSION TITLE
How i can change anchor?

### DISCUSSION THREAD

**Discussion by sheldygg** [fuma-nama/fumadocs#356] [Q&A] [answered]
Imagine what i have cyrillic in my anchor, it will be encoded if i copy-paste to some application. Can i change it?

**Comment by fuma-nama** [comment]
Not really unless you use a custom remark plugin

**Reply by sheldygg** [reply]
Can you share plugin name?

**Reply by fuma-nama** [reply]
You can search for one or implement on your own, it’s not difficult as you can take the remark plugin in this repository as a reference (`remark-heading.tsx`). It is also a documented API.

**Reply by sheldygg** [reply]
What do you mean by "you can take the remark plugin in this repository as a reference."?
I can find this function in your docs or what? If yes, can you share url to file?

**Reply by sheldygg** [reply]
Why are you running away from the question? Can you share a link to a place where it is described how to do this? Or send a link to the source code where this is done?

**Reply by fuma-nama** [reply]
Nobody, including me is paid for maintaining this open source project, please be patient. 
I don't think I have the obligation to answer every question, not to mention it's only 6 hours ago.
I'm working on lots of projects, I have limited time and might be slow to respond.
If you are simply looking for a code assistant, perhaps AI suits you better.
Every iteration takes time, please wait until the feature rolls out.

**Reply by sheldygg** [reply]
I know what that is open source project. But anyway I ask you "How i can change the anchor?", you replied what that can be implemented with some plugin, but you didn't say the name of this plugin. I try to search it in google by some keywords and find nothing what can solve my problem.

The question still remains, with which plugin I can change anchor **in your open source project**?

**Reply by fuma-nama** [reply]
This is from the first result of search: https://github.com/sindresorhus/remark-custom-header-id
Obviously, the ability of customising heading id is not yet implemented into fumadocs core, you'll have to use other remark plugins as a workarouhd. Welcome to fire a feature request in Issue tab if you want to track the progress of development.

**Reply by sheldygg** [reply]
> This is from the first result of search: https://github.com/sindresorhus/remark-custom-header-id
> Obviously, the ability of customising heading id is not yet implemented into fumadocs core, you'll have to use other remark plugins as a workarouhd. Welcome to fire a feature request in Issue tab if you want to track the progress of development.

Nice, great reply, thanks for your project!

**Comment by fuma-nama** [comment, marked as answer]
Supported in 11.0.6, use `## Heading [#slug]`

**Reply by sheldygg** [reply]
I'm sorry, but isn't working

I write it in my .mdx file `## Heading {some-head-id}` and it compiling with error. I see what your test passed this feature, so whats wrong?

![image](https://github.com/fuma-nama/fumadocs/assets/85823514/5ee40c1b-e41e-41a8-8390-fdf100476f12)

I has latest version of packages
+-- fumadocs-core@11.0.6
+-- fumadocs-mdx@8.2.15
+-- fumadocs-ui@11.0.6

**Reply by fuma-nama** [reply]
My apologies, it actually conflicts with the MDX syntax, I'll replace it with `[#slug]` later

**Reply by sheldygg** [reply]
Thanks! Waiting the fix.

**Reply by sheldygg** [reply]
See your commits, it works strange
```mdx
---
title: Test
description: Test
---

# heading1 [#slug-1]
text1

# heading2 [#slug-2]
text2
```
With this code I receive this page
![image](https://github.com/fuma-nama/fumadocs/assets/85823514/36ea7c8b-eb84-4271-94a2-eafcbdb8fad6)

**Reply by fuma-nama** [reply]
Remove your existing plugins and try again. If this doesn't work, you can open an issue for further investigation. See the [Example](https://fumadocs.vercel.app/docs/headless/mdx/toc#custom-heading-id) in the docs (also ensure you have newer versions installed)

**Reply by sheldygg** [reply]
> Удалите существующие плагины и повторите попытку. Если это не сработает, вы можете открыть проблему для дальнейшего исследования. См. [пример](https://fumadocs.vercel.app/docs/headless/mdx/toc#custom-heading-id) в документации (также убедитесь, что у вас установлены более новые версии).

https://github.com/fuma-nama/fumadocs/issues/365
